models/articles.models.js

```diff
@@ -92,3 +107,118 @@ exports.selectCommentsByArticleId = (article_id) => {
 		}
 	);
 };
+
+exports.InsertCommentByArticleId = (article_id, newComment) => {
+	const { username, body } = newComment;
+
+	if (!article_id || !username || !body) {
+		return Promise.reject({ status: 400, message: 'Bad Request' });
+	}
+
+	const fetchArticle = () => {
+		const articleQueryStr = 'SELECT * FROM articles WHERE article_id = $1;';
+		return db.query(articleQueryStr, [article_id]);
+	};
+
+	const fetchAuthor = () => {
+		const authorQueryStr = 'SELECT * FROM articles WHERE author = $1;';
+		return db.query(authorQueryStr, [username]);
+	};
+
+	return fetchArticle()
+		.then((articleResult) => {
+			if (articleResult.rows.length === 0) {
+				return Promise.reject({
+					status: 404,
+					message: 'Not Found',
+				});
+			} else {
+				return fetchAuthor();
+			}
+		})
+		.then((authorResult) => {
+			if (authorResult.rows.length === 0) {
+				return Promise.reject({
+					status: 404,
+					message: 'Not Found',
+				});
+			} else {
+				const insertComment = () => {
+					const commentsQueryStr = `INSERT INTO comments (body, article_id, author) 
+									 VALUES ($1, $2, $3) RETURNING *;`;
+					return db.query(commentsQueryStr, [
+						body,
+						article_id,
+						username,
+					]);
+				};
+
+				return insertComment().then((commentsResult) => {
+					return commentsResult.rows[0];
+				});
+			}
+		});
+};
+
+exports.updateArticleById = (votes, article_id) => {
+	if (isNaN(votes) || isNaN(article_id)) {
+		return Promise.reject({
+			status: 400,
+			message: 'Bad Request',
+		});
+	}
+	if (article_id === undefined || !votes) {
+		return Promise.reject({ status: 400, message: 'Bad Request' });
+	}
+
+	const fetchArticle = () => {
+		const articleQueryStr = 'SELECT * FROM articles WHERE article_id = $1;';
+		return db.query(articleQueryStr, [article_id]).then((result) => {
+			return result;
+		});
+	};
+
+	return fetchArticle().then((articleResult) => {
+		if (articleResult.rows.length === 0) {
+			return Promise.reject({
+				status: 404,
+				message: 'Not Found',
+			});
+		} else {
+			const updateArticle = () => {
+				const updateQueryStr = `UPDATE articles
+										SET votes = $1
+										WHERE article_id = $2 
+										RETURNING *;`;
+				return db.query(updateQueryStr, [votes, article_id]);
+			};
+
+			return updateArticle().then((result) => {
+				return result.rows[0];
+			});
+		}
+	});
+};
+
+exports.removeCommentById = (comment_id) => {
+	if (isNaN(comment_id)) {
```

Comment:
As Niamh mentioned below, we don't need this custom error handler here. PSQL should throw an error for us.